### PR TITLE
Use fake-seam-connect module to run the tests

### PR DESCRIPTION
**Pull request by andrii-balitskyi**

### CODE REVIEW

File: test/start-fake-seam-server.js
Comment (razor-x) [inline test/start-fake-seam-server.js:1]: We may not need this file if we can run the entrypoint directly: 

- https://github.com/seamapi/fake-seam-connect/blob/main/src/server.ts
- https://github.com/seamapi/fake-seam-connect/blob/main/Dockerfile#L41
- https://github.com/seamapi/fake-seam-connect/blob/main/package.json#L8

The server.js file should be included as the bin file, so if we add a script to package.json `start: fake-seam-connect --seed` it should just start the server.

The PORT can be set as an env var (really should be allowed as an arg).

IF we can remove this script it makes things better since we won't need to replicate it in the other SDKs

Comment (andrii-balitskyi) [reply test/start-fake-seam-server.js:1]: https://github.com/seamapi/python-next/pull/67/commits/6ffa7ccf045a1b92daf50b359d94c0f929ab0e05

Comment (andrii-balitskyi) [reply test/start-fake-seam-server.js:1]: With the updated approach, the test suite takes much longer time to run. Can we improve that somehow?

Comment (andrii-balitskyi) [reply test/start-fake-seam-server.js:1]: Hmm seems that it takes a lot of time only locally, on CI it takes ~30s to run the whole workflow

Comment (andrii-balitskyi) [reply test/start-fake-seam-server.js:1]: Yea, I was also thinking about port search. 

Yea, I agree. But at the same time, if it wasn't for these tests, we probably wouldn't have caught the action attempt issue so soon after that OpenAPI spec update ([relevant Slack discussion](https://hello-seam.slack.com/archives/C05QPFEPYQ7/p1716557102457209?thread_ts=1716249656.205809&cid=C05QPFEPYQ7)). Do you think we should leave them for now?

Comment (razor-x) [reply test/start-fake-seam-server.js:1]: We can leave them for now, but I do think we should generally remove them. We can selectively choose additional tests to add if we have a good reason to believe it covers a particular edge case for the SDK. For example, testing `/devices/list` and `/locks/list` is not useful, but testing `/events/list` may be important since events are non-trival union types.

For action attempts, we would still have tests using *action_attemtps/get` to test `wait_for_action_attempt.

File: package.json [hunk 1]
```diff
@@ -5,9 +5,11 @@
   "scripts": {
     "generate": "node generate-routes.js",
     "postgenerate": "make format",
-    "format": "prettier --write --ignore-path .gitignore ."
+    "format": "prettier --write --ignore-path .gitignore .",
+    "start:fake-seam-connect": "fake-seam-connect --seed"
```
Comment (razor-x) [inline package.json:9]: We can keep things simple here and use `start`. Means if we need to start more stuff later we only need to update the script here.

```suggestion
    "start": "fake-seam-connect --seed"
```

File: package.json [hunk 2]
```diff
@@ -5,9 +5,11 @@
...
   },
   "devDependencies": {
+    "@seamapi/fake-seam-connect": "^1.65.5",
```
Comment (razor-x) [inline package.json:12]: Let's pin this one as well. It will make reproducible CI easier since we won't need to check the lockfile to see what version was used. 

```suggestion
    "@seamapi/fake-seam-connect": "1.65.5",
```

File: test/conftest.py [hunk 3]
```diff
@@ -1,13 +1,47 @@
...
+import subprocess
+import time
+import os
 from seam import Seam
-import random
-import string
+from contextlib import contextmanager
+
+
+def find_free_port():
```
Comment (razor-x) [inline test/conftest.py:10]: Mainly because we already use this and people are familiar with it https://www.npmjs.com/package/get-port so we can just use one name across our projects to refer to this function.

```suggestion
def get_port():
```

**Review by razor-x — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (andrii-balitskyi) [reply package.json]: 863b945967a9bda4b6189df79686e6f0f8fc2668

File: package.json (hunk 2 above)
Comment (andrii-balitskyi) [reply package.json]: 863b945967a9bda4b6189df79686e6f0f8fc2668

File: test/conftest.py (hunk 3 above)
Comment (andrii-balitskyi) [reply test/conftest.py]: 447e37de09bf11aaff54eed1b386758e90f28a2b

File: package.json (hunk 1 above)
Comment (andrii-balitskyi) [reply package.json]: https://github.com/seamapi/python-next/pull/67/commits/9d134b3c7794cd497b61619ed1261ca9c72590e4
